MoviesTVSentiments/app/src/main/java/com/google/moviestvsentiments/service/database/AsyncDatabaseExecutor.java

```diff
@@ -0,0 +1,42 @@
+package com.google.moviestvsentiments.service.database;
+
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * A DatabaseExecutor that executes all queries on background threads.
+ */
+public class AsyncDatabaseExecutor implements DatabaseExecutor {
+
+    private final ExecutorService executor;
+
+    /**
+     * Creates an AsyncDatabaseExecutor object.
+     * @param numberOfThreads The number of threads to use when performing writes to the database.
+     */
+    private AsyncDatabaseExecutor(int numberOfThreads) {
+        executor = Executors.newFixedThreadPool(numberOfThreads);
```

Comment:
For the Room database, it seems like fixed thread pool may be holding resources when it does not use it.

https://stackoverflow.com/questions/949355/executors-newcachedthreadpool-versus-executors-newfixedthreadpool

Reply:
I changed this to a cached thread pool instead.